app/scenes/channel/channel_header/channel_header_container.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {Constants} from 'service/constants';
+import {getMyPreferences, getTheme} from 'service/selectors/entities/preferences';
+import {getCurrentUserId, getUser} from 'service/selectors/entities/users';
+import {getUserIdFromChannelName} from 'service/utils/channel_utils';
+import {displayUsername} from 'service/utils/user_utils';
+
+import ChannelHeader from './channel_header';
+
+function mapStateToProps(state, ownProps) {
+    const currentChannel = ownProps.currentChannel;
+
+    let displayName = '';
+    if (currentChannel) {
+        if (currentChannel.type === Constants.DM_CHANNEL) {
+            const otherUser = getUser(state, getUserIdFromChannelName(getCurrentUserId(state), currentChannel));
+
+            if (otherUser) {
+                displayName = displayUsername(otherUser, getMyPreferences(state));
+            }
+        } else {
+            displayName = currentChannel.display_name;
+        }
+    }
+
+    return {
```

Comment:
yes I don't get how it works if the ownProps aren't set in the return statement. but cool.

So the LHS is called channel_drawer and I think we should be consistent, I know the we are mirroring the webapp but wanted to make sure to avoid inconsistencies